apps/src/templates/manageStudents/ManageStudentsLoginInfo.jsx

```diff
@@ -15,13 +16,47 @@ class ManageStudentsLoginInfo extends Component {
     sectionId: PropTypes.number,
     sectionCode: PropTypes.string,
     loginType: PropTypes.string,
-    // The prefix for the code studio url in the current environment,
-    // e.g. '//studio.code.org' or '//localhost-studio.code.org:3000'.
-    studioUrlPrefix: PropTypes.string,
+  };
+
+  prependUrlOrigin = (url) => {
+    if (url.startsWith('//')) {
+      return `http:${url}`;
+    } else {
+      return window.location.origin + url;
+    }
+  };
+
+  joinSectionUrl = (loginType) => {
+    let path = '';
+
+    if (loginType === SectionLoginType.word || loginType === SectionLoginType.picture) {
+      path = pegasus('/join');
+    } else if (loginType === SectionLoginType.email) {
+      path = studio(`/join/${this.props.sectionCode}`);
+    }
+
+    // This URL is displayed as a raw URL to the user, so it needs to include origin (not just a relative path)
+    // if it doesn't already.
+    return this.prependUrlOrigin(path);
```

Comment:
Maybe dumb question: Are the `studio()` and `pegasus()` helpers not already including the appropriate protocol for the environment?